tests/functional/helper.js

```diff
@@ -484,6 +500,40 @@ module.exports = {
     return json.value;
   },
 
+  async resetAddons(addonPath) {
+    await this.waitForElementProperty(
+        'VPNAddonManager', 'loadCompleted', 'true');
+
+    this.resetLastAddonLoadingCompleted();
+    await this.setSetting(
+        'addon/customServerAddress', `${constants.ADDON_URL}/${addonPath}/`);
+    await this.setSetting('addon/customServer', 'true');
+
+    const json = await this._writeCommand('reset_addons');
+    assert(
+        json.type === 'reset_addons' && !('error' in json),
+        `Command failed: ${json.error}`);
+
+    await this.waitForCondition(async () => this.lastAddonLoadingCompleted());
```

Comment:
I'm confused. Does the `_writeCommand('reset_addons')` end up having `connect` called again?

Reply:
no. `_writeCommand()` writes something on a established connected channel (websocket for functional tests, wasm JS for wasm).